app/bulk_resource/views.py

```diff
@@ -92,10 +93,95 @@ def upload_data():
     db.session.commit()
     # TODO: If the only descriptors are the required ones, then no need to
     # TODO: review them.
+    reset_url = url_for('bulk_resource.review_descriptor_types')
+    update_url = url_for('bulk_resource.update_data')
+    redirect_url = reset_url if reset_or_update == 'reset' else update_url
     return jsonify(
-        redirect=url_for('bulk_resource.review_descriptor_types')
+        redirect=redirect_url
     )
 
+@bulk_resource.route('/update-data', methods=['GET'])
+@login_required
+def update_data():
+    csv_container = CsvContainer.most_recent(user=current_user)
+    if csv_container is None:
+        abort(404)
+
+    errors = []
+
+    for row in csv_container.csv_rows:
+        name = row.csv_body_cells[csv_container.name_column_index].data
+        address = \
+            row.csv_body_cells[csv_container.address_column_index].data
+        g = geocoder.google(address)
+        resource = Resource.query.filter_by(name=name).first()
+
+        if resource is None:
+            resource = Resource(
+                name=name,
+                address=address,
+                latitude=g.latlng[0],
+                longitude=g.latlng[1]
+            )
+            db.session.add(resource)
+
+        else:
+            resource.latitude = g.latlng[0]
+            resource.longitude = g.latlng[1]
+
+        for i, cell in enumerate(row.csv_body_cells):
+            if i not in csv_container.required_column_indices():
+                descriptor_name = \
+                    csv_container.csv_header_row.csv_header_cells[i].data
+                descriptor = Descriptor.query.filter_by(
+                    name=descriptor_name
+                ).first()
+                if descriptor is None:
```

Comment:
yeah this can be something we also discuss more in depth with the client @riyer15 and we can make an issue